src/compiler/utilities.ts

```diff
@@ -4181,7 +4181,31 @@ export function canHaveJSDoc(node: Node): node is HasJSDoc {
     }
 }
 
-/** @internal */
+/**
+ * Gets either the {@link JSDoc} object or {@link JSDocTag} providing documentation for the provided node.
+ * A {@link JSDoc} object will be returned if an entire comment is associated with this node. This is always

```

Comment:
I would replace this paragraph with the one below:

```
This function checks multiple locations for JSDoc comments that apply to a host node. At each location, the whole comment may apply to the node, or only a specific tag in the comment. In the first case, location adds the entire {@link JSDoc} object. In the second case, it adds the applicable {@link JSDocTag}.

For example, a JSDoc comment before a parameter adds the entire {@link JSDoc}. But a `@param` tag on the parent function only adds the {@link JSDocTag} for the `@param`.
```

(writing probably needs to be rewrapped)